src/tools/build-manifest/src/main.rs

```diff
@@ -207,21 +207,26 @@ fn main() {
     // related code in this tool and ./x.py dist hash-and-sign can be removed.
     let legacy = env::var("BUILD_MANIFEST_LEGACY").is_ok();
 
-    // Avoid overloading the old server in legacy mode.
-    if legacy {
-        rayon::ThreadPoolBuilder::new()
-            .num_threads(1)
-            .build_global()
-            .expect("failed to initialize Rayon");
-    }
+    let num_threads = if legacy {
+        // Avoid overloading the old server in legacy mode.
+        1
+    } else if let Ok(num) = env::var("BUILD_MANIFEST_NUM_THREADS") {
```

Comment:
I don't care if you don't, but this is probably best as a var_os so that we avoid the "somehow non utf-8 content snuck in and we just silently default to num_cpus::get now"

Reply:
Fixed.